specs/fulu/p2p-interface.md

```diff
@@ -204,6 +205,16 @@ The following validations MUST pass before forwarding the `sidecar: DataColumnSi
 
 *Note:* In the `verify_data_column_sidecar_inclusion_proof(sidecar)` check, for all the sidecars of the same block, it verifies against the same set of `kzg_commitments` of the given beacon block. Client can choose to cache the result of the arguments tuple `(sidecar.kzg_commitments, sidecar.kzg_commitments_inclusion_proof, sidecar.signed_block_header)`.
 
+###### Distributed Blob Publishing using blobs retrieved from local execution layer client
+
+Honest nodes SHOULD query `engine_getBlobsV2` as soon as they receive a valid `beacon_block` or `data_column_sidecar` from gossip. If ALL blobs matching `kzg_commitments` are retrieved, they should convert the response to data columns, and import the result.
```

Comment:
I've concerns on requiring it from all. If your custody requirement is just 4, it sounds like too much to follow all these rules and spend computation power. I'd bound it to supernodes only.

Reply:
Also home staker on ADSL wouldn't be happy to try to publish all DataColumns with 48, 64 etc blobs

Reply:
Full nodes won't actually send all data columns - they only send data columns on topics they're subscribed to. **This is the same behaviour as if you receive a data column via gossip** - you'd spend the bandwidth to propagate it to mesh peers.

If the node ever add the columns to the "gossip seen cache" without publishing / forwarding the column, it would stop propagation of the columns to the rest of the network.

re 1st point about computing columns - proofs will no longer be computed in the CL, so this would be a cheap operation, and potentially makes the block available sooner than waiting for columns to arrive from gossip.